### PR TITLE
aws_route53resolver_resolver_rule

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

It contains all the singular and plural resources of the above resource.

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_route53resolver_resolver_rules.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @table = fetch_data
+  end
+
+  def fetch_data
+    rows = []
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.route53resolver_client.list_resolver_rules(@query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
```
Comment (sathish-progress) [inline libraries/aws_route53resolver_resolver_rules.rb:47]: return `rows`

File: test/integration/build/aws.tf [hunk 2]
```diff
@@ -2763,10 +2763,9 @@ resource "aws_efs_file_system" "aws_efs_file_system_mt_test" {
 resource "aws_vpc" "aws_vpc_mount_mt_test" {
   cidr_block = "10.0.0.0/16"
 }
-
 resource "aws_subnet" "aws_subnet_mount_mt_test" {
   vpc_id            = aws_vpc.aws_vpc_mount_mt_test.id
-  cidr_block        = "10.0.1.0/24"
-  availability_zone = var.aws_availability_zone
+  cidr_block        = "10.0.3.0/24"
+  availability_zone = "us-east-2c"
```
Comment (sathish-progress) [inline test/integration/build/aws.tf:2769]: unless you want it specifically in this region, you can use `var.location`

**Comment by sathish-progress** [comment]
also update readme please

**Review by IanMadd — APPROVED** [review]

File: libraries/aws_route53resolver_resolver_rules.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_route53resolver_resolver_rules.rb:47]: Done

File: test/integration/build/aws.tf (hunk 2 above)
Comment (soumyo13) [reply test/integration/build/aws.tf:2769]: Done

**Review by Nirbhay1997 — APPROVED** [review]

**Pull request merged**